packages/eslint-plugin/src/rules/sort-type-union-intersection-members.ts

```diff
@@ -96,10 +96,18 @@ function getGroup(node: TSESTree.TypeNode): Group {
   }
 }
 
-function requiresParentheses(node: TSESTree.TypeNode): boolean {
+interface Constituent {
+  group: number;
+  node: TSESTree.TypeNode;
+  text: string;
+}
```

Comment:
🤔 this code is getting more complex _(which is good, it's working better!)_ and is roughly duplicated across two files. Could you extract it to a shared helper please? Maybe, `typeNodeRequiresParentheses`? `packages/eslint-plugin/src/util/` would be a good place to put it.